src/internal/runSaga.js

```diff
@@ -24,7 +25,7 @@ export function runSaga(
     subscribe,
     dispatch,
     getState,
-    {sagaMonitor, logger},
+    {sagaMonitor, logger, middleware},
```

Comment:
also aint sure if I feel naming this a `middleware` might be confusing for people, maybe it could be something more specific, like `effectMiddleware`? OTOH crippling the API to just `declarativeEffects` would be kinda crippling, why one couldnt mock other things with this?